src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -99,6 +144,7 @@ class PexRequest(EngineAwareParameter):
     repository_pex: Pex | None
     additional_args: Tuple[str, ...]
     pex_path: Tuple[Pex, ...]
+    is_lockfile: bool
```

Comment:
Should this be encoded in `PexRequirements`?

Reply:
Oh, hm maybe, yeah. I'm currently playing with a couple different factorings for a followup that will add per-tool lockfiles to the rest of the tools. Is it okay to experiment with that in the followup?